### ISSUE TITLE
Offsets in Iacono-Marziano at lower CO2 contents

### ISSUE THREAD

**Issue by PennyWieser** [kaylai/VESIcal#90]
@simonwmatthews The Iacono-Marziano code is looking good vs. the calibration dataset. 
![image](https://user-images.githubusercontent.com/65668673/85116248-f05df180-b214-11ea-9b69-5d7e8481025e.png)

I then used the web app for a bunch of compositions from the calibration dataset, and then some other tests with these major element comps but different H2O and CO2 contents. Again, mostly pretty good. 
![image](https://user-images.githubusercontent.com/65668673/85116483-4e8ad480-b215-11ea-915c-296d2b58fc1c.png)

The only issue seems to be low CO2 contents (e.g., <500 ppm) at Relatively high H2O contents (1-2 wt%)
![image](https://user-images.githubusercontent.com/65668673/85117141-4ed79f80-b216-11ea-9b1f-4285fff03d08.png)

I've added them to a folder called https://github.com/kaylai/VESIcal/tree/master/Calibration/Testing

**Comment by simonwmatthews** [comment]
I've not been able to find the origin of this so far. I checked that its not using the anhydrous vs hydrous versions of NBO/O. It doesn't seem to be- the discrepancy reverses and increases with the anhydrous version.

The puzzling thing is it seems to be split into two populations. Is there anything systematically different between the two sets of samples? Do you know if the app applies any sort of normalisation/corrections to the data?

Will look again later.

**Comment by simonwmatthews** [comment]
This is mostly a note to myself... When starting the root solver a starting guess for the fluid composition is required. If there are two regions of solution space, perhaps it's getting stuck in the wrong region for some of the samples.

**Comment by PennyWieser** [comment]
@simonwmatthews I've now added several more tabs, basically I've run  1 composition for varying H2O-CO2 contents at 1200 and 1100C - the offsets are pretty similar, suggesting its not temp. 
![image](https://user-images.githubusercontent.com/65668673/85198211-afcaaa80-b2de-11ea-8460-3e88eeca454d.png)
![image](https://user-images.githubusercontent.com/65668673/85198223-c113b700-b2de-11ea-9377-f368b4ea2569.png)

I've then run the composition that generated the big % errors yesterday through in a similar systematic way - the offsets show similar upkicks at low CO2 contents, but the magnitude is larger. Does this help trouble shoot? It might also be a bug in their online calculator I guess or strange handling of low volatile contents in either?
![image](https://user-images.githubusercontent.com/65668673/85198250-fae4bd80-b2de-11ea-8df3-d2a36861ca84.png)

**Comment by simonwmatthews** [comment]
I've got a little further with this. (And am recording my thought process here). The offsets <100% (3rd figure from top) appear to be a numerical error- likely with one of the root solvers. However, the discrepancies >100% (3rd figure from top) are a real feature of the VESIcal code. This could be an error in the webapp, or an error in how I've coded up the model (only affecting extremely high H2O contents). Probably the latter, and it might be related to how the samples are normalised?

**Comment by kaylai** [comment]
> Probably the latter, and it might be related to how the samples are normalised?

What happens if you remove `normalize_AdditionalVolatiles(sample)` or change it to FixedVolatiles or the standard normalization in the `preprocess_sample` routine?

**Comment by simonwmatthews** [comment]
Good news! Turns out the values for the H2O solubility equation given in the paper are different from those used in the online app. I've now updated the VESIcal code to use the same values as the App and the discrepancy almost disappears (< 1% at worst)

@kaylai I tried this to see if it was the cause of any of the remaining discrepancies, but it didn't seem to work... I've also tried a few other things that I thought might be causing problems, but I can't get away from there being a small remaining discrepancy.

I'm going to close this issue as the discrepancy is essentially resolved. I think the remaining discrepancy might be related to the jumps Penny described in issue #94.
